### PR TITLE
feat: add method to get decimated data to Channel

### PR DESCRIPTION
**Pull request by mariobuikhuizen**

### CODE REVIEW

File: nominal/core/channel.py [hunk 1]
```diff
@@ -114,6 +117,80 @@ def _from_conjure_logicalseries_api(
...
+            df = pd.DataFrame(
+                result.numeric.values,
+                columns=["value"],
+                index=[to_ts(timestamp) for timestamp in result.numeric.timestamps],
+            )
+            df.index.name = "timestamp"
+            return df
+
+        if result.bucketed_numeric is None:
+            raise ValueError("Unexpected response from compute service")
```
Comment (alkasm) [inline nominal/core/channel.py:153]: can you be more explicit in the error message on what is unexpected?

File: nominal/core/channel.py [hunk 2]
```diff
@@ -114,6 +117,80 @@ def _from_conjure_logicalseries_api(
...
+        Resolution in picoseconds for picosecond-granularity dataset, nanoseconds otherwise.
+        """
+        if buckets is not None:
+            # Somehow the number of points returned is buckets / 1000
+            buckets = buckets * 1000
+
+        result = self._decimate_request(start, end, buckets, resolution)
+
+        def to_ts(timestamp: api_Timestamp) -> pd.Timestamp:
+            return pd.Timestamp(timestamp.seconds, unit="s", tz="UTC") + pd.Timedelta(timestamp.nanos, unit="ns")
```
Comment (alkasm) [inline nominal/core/channel.py:140]: this doesn't create/utilize a closure so I'd just define this outside of this function

File: nominal/core/channel.py [hunk 3]
```diff
@@ -114,6 +117,80 @@ def _from_conjure_logicalseries_api(
...
 
+    def get_decimated(
+        self,
+        start: str | datetime | IntegralNanosecondsUTC,
+        end: str | datetime | IntegralNanosecondsUTC,
+        *,
+        buckets: int | None = None,
+        resolution: int | None = None,
+    ) -> pd.DataFrame:
+        """Retrieve the channel data as a pandas.Dataframe, decimated to the given buckets or resolution.
```
Comment (alkasm) [inline nominal/core/channel.py:128]:
```suggestion
        """Retrieve the channel data as a pandas.DataFrame, decimated to the given buckets or resolution.
```

File: nominal/core/channel.py [hunk 4]
```diff
@@ -114,6 +117,80 @@ def _from_conjure_logicalseries_api(
...
+
+        Enter either the number of buckets or the resolution for the output.
```
Comment (alkasm) [inline nominal/core/channel.py:130]: is it valid to provide both? if not, lets check before we make the API call?

File: nominal/core/channel.py [hunk 5]
```diff
@@ -114,6 +117,80 @@ def _from_conjure_logicalseries_api(
...
+        start: str | datetime | IntegralNanosecondsUTC,
+        end: str | datetime | IntegralNanosecondsUTC,
+        *,
+        buckets: int | None = None,
+        resolution: int | None = None,
+    ) -> pd.DataFrame:
+        """Retrieve the channel data as a pandas.DataFrame, decimated to the given buckets or resolution.
+
+        Enter either the number of buckets or the resolution for the output.
+        Resolution in picoseconds for picosecond-granularity dataset, nanoseconds otherwise.
```
Comment (alkasm) [inline nominal/core/channel.py:131]: Oof, this just silently switches behavior depending on the dataset precision? That's unfortunate if true

File: nominal/core/channel.py (hunk 4 above)
Comment (mariobuikhuizen) [reply nominal/core/channel.py:130]: Providing both results in an API error.

File: nominal/core/channel.py (hunk 5 above)
Comment (mariobuikhuizen) [reply nominal/core/channel.py:131]: It is stated like this in the docstring of SummarizeSeriesNode.resolution.

File: nominal/core/dataset.py [hunk 6]
```diff
@@ -81,6 +82,8 @@ def logical_series(self) -> timeseries_logicalseries.LogicalSeriesService: ...
         def upload(self) -> upload_api.UploadService: ...
         @property
         def units(self) -> scout.UnitsService: ...
+        @property
+        def compute(self) -> scout_compute_api.ComputeService: ...
```
Comment (alkasm) [inline nominal/core/dataset.py:86]: ah, this type should inherit from `Channel._Clients` and then it would propagate automatically

Comment (alkasm) [reply nominal/core/dataset.py:86]: Runs does this as well like so:

```py
class _Clients(Dataset._Clients, Attachment._Clients, HasAuthHeader, Protocol):
```

note that you need to put the inheritance _before_ `HasAuthHeader`

**Review by alkasm — APPROVED** [review]

**Pull request merged**